packages/data/src/components/with-dispatch/test/index.js

```diff
@@ -31,9 +36,15 @@ describe( 'withDispatch', () => {
 
 	it( 'passes the relevant data to the component', async () => {
 		const user = userEvent.setup();
+		const buttonSpy = jest.fn();
 		const registry = createRegistry();
 		registry.registerStore( 'counter', storeOptions );
 
+		const Button = memo( ( { onClick } ) => {
+			buttonSpy();
+			return <button onClick={ onClick } />;
+		} );
```

Comment:
It's also possible, and elegant, to wrap the component function itself in a spy:
```js
const ButtonSpy = jest.fn( ( { onClick } ) => <button onClick={ onClick } /> );
const Button = memo( ButtonSpy );
```
The `useSelect` tests use this pattern to count renders.

Reply:
Oh, that's a neat one. Thanks, @jsnajdr!